### PR TITLE
Add matmul microbenchmark.

### PR DESCRIPTION
**Pull request by golechwierowicz**
With the pure wall time infrastructure came nice
ability to do microbenchmarks easily. I added
the sample microbenchmark for matmul.

Additionally for slower runs it dumps PT profiles/traces of base, and test functions.

### CODE REVIEW

**Comment by vanbasten23** [comment]
Why not merging to the master branch?

Also, can you add the feature to the readme?

**Comment by golechwierowicz** [comment]
> Why not merging to the master branch?
> 
> Also, can you add the feature to the readme?

Hey, it depends on changes in the previous branch, bench.py file which provides necessary methods to make this microbenchmark work.

I'll add the usage instructions in the next PR.

**Comment by cota** [comment]
Note that until the baseline is the master branch, CI will not run.
I encountered this in the past, and couldn't get CI to run even after resetting the baseline to the master branch. I ended up creating a new PR based off master so that CI would run.

File: benchmarks/microbench.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  return MicrobenchResults(
+      test_name=test_name,
+      testing_speedup=baseline_wall_ms / testing_wall_ms,
+      baseline_wall_ms=baseline_wall_ms,
+      testing_wall_ms=testing_wall_ms,
+  )
+
+
+def _dump_flame(filepath, prof):
+  """Data can be interpreted with running ./flamegraph.pl --title "CUDA time" --countname "us." /tmp/profiler_stacks.txt > perf_viz.svg"""
```
Comment (cota) [inline benchmarks/microbench.py:96]: Where does `flamegraph.pl` come from?

Comment (golechwierowicz) [reply benchmarks/microbench.py]: Good point, I added the comment mentioning how to do a set up.

**Review by cota — APPROVED** [review]

**Pull request merged**